dkg-gadget/src/worker.rs

```diff
@@ -1257,23 +1247,63 @@ where
 		}
 	}
 
+	/// Get unsigned proposals and create offline stage using an encoded (chainId, dkgpayloadKey) as the round key
+	fn create_offline_stages(&mut self, header: &B::Header) {
+		if self.rounds.is_none() {
+			return
+		}
+
+		let at = BlockId::hash(header.hash());
+		let unsigned_proposals = match self.client.runtime_api().get_unsigned_proposals(&at) {
+			Ok(res) => res,
+			Err(_) => return,
+		};
+
+		let rounds = self.rounds.as_mut().unwrap();
+		let mut errors = vec![];
+		if rounds.is_offline_ready() {
+			for (key, ..) in &unsigned_proposals {
+				if self.dkg_state.reset_signers_at.contains_key(&key.encode()) {
+					continue
+				}
+				// TODO: use deterministic random signers set
+				let signer_set_id = rounds.get_id();
+				let s_l = (1..=rounds.dkg_params().2).collect();
+				match rounds.reset_signers(key.encode(), signer_set_id, s_l, *header.number()) {
```

Comment:
More comments here necessary. Is this using the on-chain threshold of signers?

Reply:
Ensure we define whether we use:
- the same signing set for all offline stages within an epoch
- rotate it based on the `(ChainId, DKGPayloadKey)` deterministically

Reply:
Consider choosing the signing set once each time the authority set changes. Ensure that we are using the on-chain defined signing threshold.